zkevm-circuits/src/evm_circuit/execution/selfbalance.rs

```diff
@@ -0,0 +1,188 @@
+use crate::{
+    evm_circuit::{
+        execution::ExecutionGadget,
+        step::ExecutionState,
+        table::{AccountFieldTag, CallContextFieldTag},
+        util::{
+            common_gadget::SameContextGadget,
+            constraint_builder::{ConstraintBuilder, StepStateTransition, Transition::Delta},
+            Cell, Word,
+        },
+        witness::{Block, Call, ExecStep, Transaction},
+    },
+    util::Expr,
+};
+use eth_types::{ToLittleEndian, ToScalar};
+use halo2::{arithmetic::FieldExt, circuit::Region, plonk::Error};
+
+#[derive(Clone, Debug)]
+pub(crate) struct SelfbalanceGadget<F> {
+    same_context: SameContextGadget<F>,
+    callee_address: Cell<F>,
+    self_balance: Word<F>,
+}
+
+impl<F: FieldExt> ExecutionGadget<F> for SelfbalanceGadget<F> {
+    const NAME: &'static str = "SELFBALANCE";
+
+    const EXECUTION_STATE: ExecutionState = ExecutionState::SELFBALANCE;
+
+    fn configure(cb: &mut ConstraintBuilder<F>) -> Self {
+        let callee_address = cb.query_cell();
+        cb.call_context_lookup(
+            None,
+            CallContextFieldTag::CalleeAddress,
+            callee_address.expr(),
+        );
+
+        let self_balance = cb.query_rlc();
```

Comment:
Considering that the `self_balance` value is only used in the RLC encoded form in this gadget, maybe we can avoid using `Word` (which requires 32 cells), and just use a `Cell` for `self_balance`?
I think the current implementation is correct nevertheless.

Reply:
updated the PR to use `Cell` instead.